src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -544,8 +544,7 @@ void ReplicatedMergeTreeQueue::removeProcessedEntry(zkutil::ZooKeeperPtr zookeep
     if (!found && need_remove_from_zk)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Can't find {} in the memory queue. It is a bug. Entry: {}",
                                                       entry->znode_name, entry->toString());
-
-    notifySubscribers(queue_size);
+    notifySubscribers(queue_size, entry->znode_name);
```

Comment:
Good, of course you are right. Sometimes replica can decide to push something into own queue without log.